### DISCUSSION TITLE
chunked video upload?

### DISCUSSION THREAD

**Discussion by tvierb** [semifor/Twitter-API#28] [Q&A] [answered]
Hi,

is it possible to do a chunked video upload with upload_media()? Is this supported?
I neither found it in the docs nor in the source (searching for "video" "segmented" "chunk").

thanks
tvierb

**Comment by semifor** [comment, marked as answer]
Hi @tvierb,

I had actually never tried a chunked upload before. Your question prompted me to do it. I'm not sure chunked upload was available when I initially wrote the `upload_media` support. You have to make multiple `upload_media` calls with commands INIT, APPEND, FINALIZE, and STATUS. That final call, with the current code, is a bit more verbose because `upload_media` is defined as a POST while the STATUS command requires a GET request.

I successfully uploaded a large video with this code: [chunked-media-upload.pl](https://gist.github.com/semifor/d902b13118b25706ab9d613d163e13f3). (Well, the upload itself was successful, but Twitter rejected it because it was over the maximum time limit. Nonetheless, I got a good, final STATUS response so I'm confident the example code works.)

It's likely not optimum, but it's the first working draft. For some reason, I couldn't get the `media` parameter working with raw, binary data for the APPEND command. I resorted to base64 encoded data using the `media_data` parameter instead.

I relied on Twitter's [chunked media upload documentation](https://developer.twitter.com/en/docs/twitter-api/v1/media/upload-media/uploading-media/chunked-media-upload) and the @twitterdev [python example](https://github.com/twitterdev/large-video-upload-python).

I'll likely add a more refined version of this in [examples](https://github.com/semifor/Twitter-API/tree/master/examples) in an upcoming release.
